tests/Unit/Core/Form/IdentifiableObject/CommandBuilder/Product/ShippingCommandBuilderTest.php

```diff
@@ -31,27 +31,90 @@
 use Generator;
 use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductShippingCommand;
 use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\DeliveryTimeNoteType;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
 use PrestaShop\PrestaShop\Core\Form\IdentifiableObject\CommandBuilder\Product\ShippingCommandsBuilder;
 
 class ShippingCommandBuilderTest extends AbstractProductCommandBuilderTest
 {
     /**
-     * @dataProvider getExpectedCommands
+     * @dataProvider getExpectedSingleShopCommands
      *
      * @param array $formData
      * @param array $expectedCommands
      */
-    public function testBuildCommand(array $formData, array $expectedCommands): void
+    public function testBuildSingleShopCommands(array $formData, array $expectedCommands): void
     {
-        $builder = new ShippingCommandsBuilder();
-        $builtCommands = $builder->buildCommands($this->getProductId(), $formData);
+        $builder = new ShippingCommandsBuilder(self::MODIFY_ALL_SHOPS_PREFIX);
+        $builtCommands = $builder->buildCommands($this->getProductId(), $formData, $this->getSingleShopConstraint());
         $this->assertEquals($expectedCommands, $builtCommands);
     }
 
+    /**
+     * @dataProvider getExpectedAllShopCommands
+     *
+     * @param array $formData
+     * @param array $expectedCommands
+     */
+    public function testBuildAllShopCommands(array $formData, array $expectedCommands): void
+    {
+        $builder = new ShippingCommandsBuilder(self::MODIFY_ALL_SHOPS_PREFIX);
+        $builtCommands = $builder->buildCommands($this->getProductId(), $formData, $this->getSingleShopConstraint());
+        $this->assertEquals($expectedCommands, $builtCommands);
+    }
+
+    /**
+     * @return Generator
+     */
+    public function getExpectedAllShopCommands(): Generator
+    {
+        $command = $this->getAllShopsCommand();
+        /* @phpstan-ignore-next-line */
```

Comment:
It's weird that you need to ignore this line no? Since the setters expects a `int[]` parameter it should match

Reply:
yeah probably a mistake